changelog/unreleased/enhancement-add-custom-search-service

```diff
@@ -0,0 +1,7 @@
+Enhancement: Add custom search service
+
+We've added `search` as another core app that can be utilized by 
```

Comment:
@fschade is this (still/already) true? Think we were missing some functionality in order to being able to allow 3rd parties to provide their own search, right?

Reply:
yup, we decided to ship everything now and improve over time